データベース設計仕様

tossy9 opened the pull request:
# What
データベース設計仕様

# Why
データベース実装に必要なため

Commit:
create pull request

Commit:
add data base design spec

add data base design spec in README.md
define some tables and association

# Tables
- users
- groups
- members
- messages

miyagit commented on README.md:
```diff
@@ -5,20 +5,66 @@ application up and running.
+|image|string||
+|group_id|integer|null: false, foreign_key: true|
+|user_id|integer|null: false, foreign_key: true|
+`foreign_key: true` option includes `index: true` option
```
そのままテーブルの形で書いてもらって大丈夫ですよ。

miyagit commented on README.md:
```diff
@@ -5,20 +5,66 @@ application up and running.
+
+#### Association
+- belongs_to :user
+- has_many :members, :messages
```
この書き方は変ですね。

miyagit commented on README.md:
```diff
@@ -5,20 +5,66 @@ application up and running.
 
-* Services (job queues, cache servers, search engines, etc.)
+#### Association
+- has_many :groups, :members, :messages
```
ここもそうですね。

miyagit commented on README.md:
```diff
@@ -5,20 +5,66 @@ application up and running.
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, index: true|
+|email|string|null: false, unique: true|
```
deviseで追加されるカラムは入れなくて大丈夫です。

Commit:
modify data base design

## users table
- delete columns:  email, password
- add `unique: true` to name
- modify association description

## groups table
- modify association description

## members table
- add `index: true` option
- modify association description

## messages table
- add `index: true` option
- modify association description

tossy9 commented:
レビューのご指摘事項に対応しました。

ta910 commented on README.md:
```diff
@@ -5,20 +5,69 @@ application up and running.
+#### Association
 
-* Deployment instructions
+- has_many :groups
```
中間テーブル経由させましょう、throughを用います

ta910 commented on README.md:
```diff
@@ -5,20 +5,69 @@ application up and running.
+
+#### Association
+
+- belongs_to :user
```
groupは１人のユーザーにのみ属しているのでしょうか

Commit:
modify association between users and groups

## users table
- add `through: :members`

## groups table
- change `belongs_to` to `has_many`
- add  `through: :members`

tossy9 commented:
指摘事項に対応しました。

no-iwamatsu-tech commented on README.md:
```diff
@@ -5,20 +5,69 @@ application up and running.
 
-* Database creation
+## Database creation
 
```
これより上の今回使わない部分は削除してもらって大丈夫ですよ。

no-iwamatsu-tech commented on README.md:
```diff
@@ -5,20 +5,69 @@ application up and running.
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, unique: true, index: true|
 
```
deviseで標準的に用意されるんですけど、emailとpasswordは追加してください！

no-iwamatsu-tech commented on README.md:
```diff
@@ -5,20 +5,69 @@ application up and running.
+- belongs_to :user
+- belongs_to :group
+
+## Database initialization
```
ここより下の部分も削除して大丈夫ですよ！

Commit:
add email and password to users table

add two column created by gem `devise`
 - email
 - password

Commit:
delete unnecessary description

Commit:
change heading level created with Markdown

tossy9 commented:
指摘事項に対応しました。
deviseで生成されるemail, passwordは一度削除しましたが、記述を戻しました。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,55 @@
 
-* System dependencies
+- has_many :groups, through: :members
+- has_many :members
```
membersを最初に記述しましょう。
has_many :groups, through: :membersを先に記述してしまうとgroupsとusersの多々多が認識されない場合があるので。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,55 @@
 
-* How to run the test suite
+- has_many :user, through: :members
+- has_many :members
```
こちらも、です。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,55 @@
+|name|string|null: false, unique: true, index: true|
+|email|string|null: false, unique: true|
+|password|string|null: false|
+ * the gem `devise` creates email and password
```
この記述、僕はいいと思います👍

MrUrakou reviewed (commented):
idはreference型を使用しましょう。
integer型で定義してその後indexを貼ってもよいですが、記述量も増えるため定義時にreference型かつforeign_key :trueかつnull: falseをつけたほうが良いです！
参考url
https://qiita.com/ryouzi/items/2682e7e8a86fd2b1ae47

Commit:
change the association order of members

Because many-to-many relationships may not be recognized, change the
association order of members

Commit:
change contents of tables named members and messages

change `integer` to `reference` about group_id and user_id

tossy9 commented:
指摘事項対応しました。

tossy9 commented:
レビュー合格したためマージします
